### PR TITLE
Adds a rule to check that the name of a windows virtual machine is valid

### PR DESCRIPTION
**Pull request by JannoTjarks**
Hi,

i added the rule `azurerm_windows_virtual_machine_invalid_name`.

It checks, if the name is a valid windows server hostname, based on this documenation:
https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftcompute
image

Based on the terraform documentation, the check is skipped, if the `computer_name`  attribute is specified. In future this rule could be expanded or a second rule specific for `computer_name could be added`.
image

Thank you!

### CODE REVIEW

**Comment by JannoTjarks** [comment]
@wata727 could you help me with the failing check?

**Comment by wata727** [comment]
CI is failing due to differences with automatically generated code.
Since `docs/README.md` is automatically generated, please update [the template](https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/85dd9b75f911a01fd7341cc2547345fe4f495f2d/tools/apispec-rule-gen/doc_README.md.tmpl) and commit the generated code.

**Comment by JannoTjarks** [comment]
> CI is failing due to differences with automatically generated code. Since `docs/README.md` is automatically generated, please update [the template](https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/85dd9b75f911a01fd7341cc2547345fe4f495f2d/tools/apispec-rule-gen/doc_README.md.tmpl) and commit the generated code.

@wata727, got it, thanks for the hint :)
Now all checks are green. I hope, it looks good to you!

File: rules/azurerm_windows_virtual_machine_invalid_name.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		primaryAttribute, exists := resource.Body.Attributes[r.attributeName]
+		if !exists {
+			continue
+		}
+
+		err := runner.EvaluateExpr(primaryAttribute.Expr, func(val string) error {
+			if !r.pattern.MatchString(val) {
+				runner.EmitIssue(
+					r,
+					fmt.Sprintf(`"%s" does not match valid pattern %s`, val, `^[a-zA-Z0-9]{0,1}[a-zA-Z0-9]{0,13}[a-zA-Z0-9]$`),
```
Comment (wata727) [inline rules/azurerm_windows_virtual_machine_invalid_name.go:78]: It seems that the regexp here and the regexp actually checked are different. Is this correct?

```suggestion
					fmt.Sprintf(`"%s" does not match valid pattern %s`, val, `^[a-zA-Z0-9]{0,1}[a-zA-Z0-9-]{0,13}[a-zA-Z0-9]$`),
```

**Review by wata727 — APPROVED** [review]
Thanks!

**Pull request merged**